### ISSUE TITLE
pyparsing 2.4.3 breaks PyContracts

### ISSUE THREAD

**Issue by nedbat** [pyparsing/pyparsing#161]
I use PyContracts as part of testing coverage.py.  This morning, I started getting failures like this:
```
/Users/ned/coverage/trunk/.tox/py37/lib/python3.7/site-packages/contracts/library/extensions.py:217: SyntaxWarning: Exception creating Regex for oneOf, building MatchFirst
  identifier_expression = (Combine(oneOf(list(alphas)) + Word('_' + alphanums)) +
/Users/ned/coverage/trunk/.tox/py37/lib/python3.7/site-packages/contracts/library/suggester.py:60: SyntaxWarning: Exception creating Regex for oneOf, building MatchFirst
  pattern = Combine(oneOf(list(alphas)) + Word('_' + alphanums))
/Users/ned/coverage/trunk/.tox/py37/lib/python3.7/site-packages/contracts/library/variables.py:91: SyntaxWarning: Exception creating Regex for oneOf, building MatchFirst
  int_variables = (oneOf(alphabetu.split())
/Users/ned/coverage/trunk/.tox/py37/lib/python3.7/site-packages/contracts/library/variables.py:92: SyntaxWarning: Exception creating Regex for oneOf, building MatchFirst
  + FollowedBy(NotAny(oneOf(nofollow.split()))))
/Users/ned/coverage/trunk/.tox/py37/lib/python3.7/site-packages/contracts/library/variables.py:93: SyntaxWarning: Exception creating Regex for oneOf, building MatchFirst
  misc_variables = (oneOf(alphabetl.split())
/Users/ned/coverage/trunk/.tox/py37/lib/python3.7/site-packages/contracts/library/variables.py:94: SyntaxWarning: Exception creating Regex for oneOf, building MatchFirst
  + FollowedBy(NotAny(oneOf(nofollow.split() + ['x']))))
/Users/ned/coverage/trunk/.tox/py37/lib/python3.7/site-packages/contracts/library/variables.py:109: SyntaxWarning: Exception creating Regex for oneOf, building MatchFirst
  int_variables2 = (oneOf(alphabetu.split())
/Users/ned/coverage/trunk/.tox/py37/lib/python3.7/site-packages/contracts/library/variables.py:110: SyntaxWarning: Exception creating Regex for oneOf, building MatchFirst
  + FollowedBy(NotAny(oneOf(nofollow.split()))))
/Users/ned/coverage/trunk/.tox/py37/lib/python3.7/site-packages/contracts/library/variables.py:111: SyntaxWarning: Exception creating Regex for oneOf, building MatchFirst
  misc_variables2 = (oneOf(alphabetl.split())
/Users/ned/coverage/trunk/.tox/py37/lib/python3.7/site-packages/contracts/library/variables.py:112: SyntaxWarning: Exception creating Regex for oneOf, building MatchFirst
  + FollowedBy(NotAny(oneOf(nofollow.split() + ['x']))))
```
It works fine with pyparsing 2.4.2.  I don't know if this is a regression in pyparsing, or a long-standing incorrect use by PyContracts.

**Comment by nedbat** [comment]
To see what I saw:
```
# Using Python == 3.6
$ git clone https://github.com/nedbat/coveragepy
$ cd coveragepy
$ git checkout 1ecf86ee
$ pip install tox
$ tox -e py36 -- -n 0 -k zero_coverage_while_loop
```

**Comment by ptmcg** [comment]
This is easily reproduced with this expression:

    oneOf(list(alphas))

I am completely baffled that I don't have a unit test containing this common usage of `oneOf`. Thanks for the quick post!

**Comment by ptmcg** [comment]
Warnings from unresolved symbol reference were masked in the unit tests by stdout buffering. Fixed for now in 2.4.4 (and fixed the unresolved symbol too), changes will be subsumed into 3.0.0 refactoring.

**Comment by ptmcg** [comment]
Fixed in release 2.4.4
